presto-spi/src/main/java/io/prestosql/spi/predicate/SortedRangeSet.java

```diff
@@ -229,8 +229,8 @@ public SortedRangeSet intersect(ValueSet other)
 
         Builder builder = new Builder(type);
 
-        Iterator<Range> iterator1 = getOrderedRanges().iterator();
```

Comment:
It might be enough to modify `io.prestosql.spi.predicate.SortedRangeSet#getOrderedRanges` to return `unmodifiableList(lowIndexedRanges.values())`. There is no reason to return `new ArrayList<>(...)`, is there?

Reply:
`lowIndexedRanges.values()` is a `Collection`, but `getOrderedRanges()` returns a `List`

Reply:
I missed that. Maybe ...

```java
/**
 * Returns the ranges. Iteration order on the returned collection is matches the ordering of the ranges. 
 */
@JsonProperty("ranges")
public Collection<Range> getOrderedRanges()
{
    return unmodifiableCollection(lowIndexedRanges.values());
}
```

Reply:
`getOrderedRanges` is expected to return an ordered collection. `Collection` doesn't convey that. I'd also make it asymmetric vs the constructor, which takes a `List`. In any case, this is the public interface of the class. The call sites I changed are all private details of the class.

Reply:
> `Collection` doesn't convey that

This is why i added a javadoc.
Anyway -- I am fine with your approach (I will approve the PR).

However, I doubt users of `getOrderedRanges()` do want the copying behavior, so there is some opportunity here.